In [1]:
import pandas as pd
import numpy as np
import warnings
warnings.filterwarnings("ignore")

In [3]:
# Load the dataset from a CSV file
asthma_df = pd.read_csv("asthma_data_2016.csv")

# Display the first 15 rows of the DataFrame to inspect the data
asthma_df.head(15)

DUID  PID  DUPERSID  CONDN       CONDIDX  PANEL  CONDRN  AGEDIAG  CRND1  \
0   10006  101  10006101     21  100061010021     21       1       10      0   
1   10008  102  10008102     11  100081020011     21       1        1      1   
2   10008  106  10008106     11  100081060011     21       1        3      1   
3   10019  102  10019102     51  100191020051     21       1       12      1   
4   10034  101  10034101    121  100341010121     21       3       -8      0   
5   10079  101  10079101     61  100791010061     21       1        3      1   
6   10085  101  10085101     21  100851010021     21       1        1      1   
7   10085  103  10085103     11  100851030011     21       1        1      1   
8   10091  103  10091103     31  100911030031     21       1       22      0   
9   10092  102  10092102     51  100921020051     21       1       -8      0   
10  10092  105  10092105     12  100921050012     21       1        0      0   
11  10107  101  10107101     11  101071010011     21       1       24      0   
12  10114  101  10114101     31  101141010031     21       1        4      0   
13  10116  101  10116101     11  101161010011     21       1       10      1   
14  10135  101  10135101     41  101351010041     21       1        5      0   

    CRND2  ...  RXOPR16  RXOPU16  RXOSR16  RXPTR16  RXOTH16      FAMWT16F  \
0       1  ...        0        0        0        0        0  46571.054815   
1       0  ...        0        0        0        0        0  11429.276863   
2       0  ...        0        0        0        0        0  11429.276863   
3       1  ...        0        0        0     5513        0   9122.447495   
4       0  ...        0        0        0        0        0   2013.580601   
5       1  ...        0        0        0        0        0  10894.051929   
6       0  ...        0        0        0        0        0   2877.269578   
7       0  ...        0        0        0        0       16   2877.269578   
8       1  ...        0        0        0     8825        0   8748.242230   
9       1  ...        0        0        0      248        0   3938.414568   
10      1  ...        0        0        0        0        0   3938.414568   
11      0  ...        0        0        0       35        0  11500.045265   
12      1  ...        0        0        0     1279        0      0.000000   
13      0  ...        0        0        0     4073        0  12594.760774   
14      1  ...        0        0        0        0        0  48346.416561   

        FAMWT16C      SAQWT16F     DIABW16F  CSAQW16F  
0   46571.054815  36022.986856      0.00000       0.0  
1   11429.276863  17211.768075      0.00000       0.0  
2   11429.276863      0.000000      0.00000       0.0  
3    9122.447495  12874.498259      0.00000       0.0  
4    2013.580601      0.000000      0.00000       0.0  
5   10894.051929  12952.023899  15425.32206       0.0  
6    2877.269578   3117.062822      0.00000       0.0  
7    2877.269578      0.000000      0.00000       0.0  
8    5856.618147   6349.476405      0.00000       0.0  
9    3938.414568   3246.221530      0.00000       0.0  
10   3938.414568   4507.930842      0.00000       0.0  
11  11500.045265  11873.988894      0.00000       0.0  
12      0.000000  14003.257380      0.00000       0.0  
13  12594.760774  14857.752770      0.00000       0.0  
14  48346.416561  51714.760786      0.00000       0.0  

[15 rows x 1959 columns]

In [4]:
# Get the number of rows (tuples) and columns (attributes) in the dataset
rows, columns = asthma_df.shape
print(f"Number of rows (tuples): {rows}")
print(f"Number of columns (attributes): {columns}")

Number of rows (tuples): 2254
Number of columns (attributes): 1959


In [5]:
# Access and print the value at a specific location (row 1494, column 229) in the DataFrame
# Using .iloc to access the value by index
value = asthma_df.iloc[1494,229]
print(value)

4


In [6]:
# Select a subset of columns that are relevant for further analysis
selected_cols = ["DUPERSID", "SEX", "ASPRIN53", "ADAPPT42", "ADHECR42", "AGE16X", "BMINDX53", "CHBMIX42", "FAMINC16", "WAGEP16X", "TTLP16X", "UNEMP16X", "RACETHX", "TOTEXP16"]

# Create a new DataFrame with only the selected columns
asthma_selected_df = asthma_df[selected_cols]

# Display the first 15 rows of the new DataFrame with the selected columns
asthma_selected_df.head(15)

DUPERSID  SEX  ASPRIN53  ADAPPT42  ADHECR42  AGE16X  BMINDX53  CHBMIX42  \
0   10006101    2         2         2         9      43      20.0      -1.0   
1   10008102    1         2         0        -1      19      20.7      -1.0   
2   10008106    2        -1        -1        -1       9      -1.0      -9.0   
3   10019102    2         1         4         8      73      30.5      -1.0   
4   10034101    1         2        -1        -1      75      34.9      -1.0   
5   10079101    2         2         6         8      67      28.2      -1.0   
6   10085101    2         2         0        -1      44      28.1      -1.0   
7   10085103    1        -1        -1        -1      15      -1.0      -9.0   
8   10091103    2         2         5        10      36      23.4      -1.0   
9   10092102    2         1         3         7      48      29.1      -1.0   
10  10092105    2         2         2         9      22      22.7      -1.0   
11  10107101    1         2         2         9      32      21.3      -1.0   
12  10114101    2         2         4         7      47      25.1      -1.0   
13  10116101    2         2         6         7      36      29.0      -1.0   
14  10135101    1         1         4         8      68      23.5      -1.0   

    FAMINC16  WAGEP16X  TTLP16X  UNEMP16X  RACETHX  TOTEXP16  
0        0.0       0.0      0.0         0        2        23  
1    45958.0   11432.0  11432.0         0        2      3418  
2    45958.0       0.0      0.0         0        2      1517  
3    55921.0       0.0  15921.0         0        2     15794  
4    13332.0       0.0  13332.0         0        3     10340  
5    28447.0       0.0  28447.0         0        2     36677  
6    20462.0   10231.0  10231.0         0        1       216  
7    20462.0       0.0      0.0         0        1      1420  
8    21981.0   21981.0  21981.0         0        2     38353  
9    35720.0       0.0      0.0         0        1      3816  
10   35720.0    8400.0  16800.0         0        1      3584  
11   51100.0   50000.0  51100.0         0        2       185  
12  118727.0   29682.0  29682.0         0        1      3581  
13  112080.0   22400.0  22440.0         0        2      8991  
14   73307.0       0.0  32633.0         0        2     10656

In [7]:
# Select a subset of rows and columns from the DataFrame
# loc[150:160] selects rows with index labels 150 to 160 (inclusive)
# ['DUPERSID', 'SEX'] selects only the 'DUPERSID' and 'SEX' columns from these rows
subset = asthma_selected_df.loc[150:160,['DUPERSID','SEX']]
print(subset)

     DUPERSID  SEX
150  11461204    1
151  11465102    2
152  11465103    1
153  11465105    2
154  11477102    1
155  11479103    2
156  11480102    1
157  11500103    2
158  11508103    1
159  11527101    2
160  11546101    2


In [8]:
# Describe
asthma_selected_df.describe(include="all")

DUPERSID          SEX     ASPRIN53     ADAPPT42     ADHECR42  \
count  2.254000e+03  2254.000000  2254.000000  2254.000000  2254.000000   
mean   4.635805e+07     1.594055     0.807897     0.876664     3.052351   
std    3.031418e+07     0.491183     1.505401     3.417645     5.352129   
min    1.000610e+07     1.000000    -8.000000    -9.000000    -9.000000   
25%    1.500510e+07     1.000000    -1.000000    -1.000000    -1.000000   
50%    7.047910e+07     2.000000     2.000000     0.000000    -1.000000   
75%    7.551310e+07     2.000000     2.000000     4.000000     8.000000   
max    8.056810e+07     2.000000     2.000000     6.000000    10.000000   

            AGE16X     BMINDX53     CHBMIX42       FAMINC16       WAGEP16X  \
count  2254.000000  2254.000000  2254.000000    2254.000000    2254.000000   
mean     36.867791    20.215129     3.085226   55888.935670   13765.931233   
std      24.034232    16.336593     9.687313   58137.780662   28348.237042   
min      -1.000000    -9.000000    -9.000000  -48049.000000       0.000000   
25%      13.000000    -1.000000    -1.000000   16796.000000       0.000000   
50%      36.000000    25.100000    -1.000000   37477.500000       0.000000   
75%      57.000000    32.000000    -1.000000   77417.750000   16000.000000   
max      85.000000    89.800000    57.200000  495465.000000  221815.000000   

             TTLP16X     UNEMP16X      RACETHX       TOTEXP16  
count    2254.000000  2254.000000  2254.000000    2254.000000  
mean    19562.167258    35.235138     2.209406    9278.631322  
std     30358.241468   378.795652     1.076143   20089.432125  
min    -44824.000000     0.000000     1.000000       0.000000  
25%         0.000000     0.000000     1.000000     881.000000  
50%      8820.000000     0.000000     2.000000    2904.000000  
75%     26150.000000     0.000000     3.000000    9135.750000  
max    228314.000000  5560.000000     5.000000  417922.000000

In [10]:
asthma_selected_df.columns

Index(['DUPERSID', 'SEX', 'ASPRIN53', 'ADAPPT42', 'ADHECR42', 'AGE16X',
       'BMINDX53', 'CHBMIX42', 'FAMINC16', 'WAGEP16X', 'TTLP16X', 'UNEMP16X',
       'RACETHX', 'TOTEXP16'],
      dtype='object')

In [12]:
asthma_selected_df['SEX'].value_counts()

SEX
2    1339
1     915
Name: count, dtype: int64

In [14]:
# Get value counts for each column
value_counts = {col: asthma_selected_df[col].value_counts() for col in asthma_selected_df.drop(['DUPERSID'],axis=1).columns}

# Display the result
for col, counts in value_counts.items():
    print(f"Value counts for {col}:\n{counts}\n")

Value counts for SEX:
SEX
2    1339
1     915
Name: count, dtype: int64

Value counts for ASPRIN53:
ASPRIN53
 2    1140
-1     718
 1     380
-8       9
-7       7
Name: count, dtype: int64

Value counts for ADAPPT42:
ADAPPT42
-1    880
 5    243
 6    188
 0    183
 2    177
 3    158
 1    156
 4    150
-9    119
Name: count, dtype: int64

Value counts for ADHECR42:
ADHECR42
-1     1063
 10     332
 8      248
 9      194
 7      155
-9      103
 5       60
 6       48
 4       26
 2        8
 3        8
 0        6
 1        3
Name: count, dtype: int64

Value counts for AGE16X:
AGE16X
 9     64
 10    59
 11    54
 7     51
 13    49
       ..
 81     9
 83     7
-1      7
 84     7
 80     7
Name: count, Length: 87, dtype: int64

Value counts for BMINDX53:
BMINDX53
-1.0     718
 26.6     28
 25.8     25
 28.3     25
 30.0     25
        ... 
 39.8      1
 16.6      1
 17.9      1
 48.9      1
 31.7      1
Name: count, Length: 320, dtype: int64

Value counts for CHBMIX42:
CHBMIX42
-

In [15]:
# Number of Nulls before converting negative values to nulls
asthma_selected_df.isnull().sum()

DUPERSID    0
SEX         0
ASPRIN53    0
ADAPPT42    0
ADHECR42    0
AGE16X      0
BMINDX53    0
CHBMIX42    0
FAMINC16    0
WAGEP16X    0
TTLP16X     0
UNEMP16X    0
RACETHX     0
TOTEXP16    0
dtype: int64

In [16]:
# Replace negative values with NaN
asthma_selected_df = asthma_selected_df.where(asthma_selected_df >= 0, np.nan)

In [17]:
# Number of Nulls before converting negative values to nulls
asthma_selected_df.isnull().sum()

DUPERSID       0
SEX            0
ASPRIN53     734
ADAPPT42     999
ADHECR42    1166
AGE16X         7
BMINDX53     738
CHBMIX42    1808
FAMINC16       1
WAGEP16X       0
TTLP16X        1
UNEMP16X       0
RACETHX        0
TOTEXP16       0
dtype: int64

CHBMIX42    1808 , Has most missing values

SEX: As it represents gender and has two categories MALE and FEMALE.
RACETHX: Represents race/ethnicity categories.
ASPRIN53: As per the code book there are two values i.e., 1, 2 which represent Yes and No respectively. So, this is considered categorical.

In [18]:
# Check the updated column before mean imputation
print("before mean imputation null count : ",asthma_selected_df['ADHECR42'].isnull().sum())

# ADHECR42 mean
mean_adhecr42 = asthma_selected_df['ADHECR42'].mean()

# Fill nulls with mean value
asthma_selected_df['ADHECR42'].fillna(mean_adhecr42, inplace=True)

# Check the updated column after mean imputation
print("after mean imputation null count : ",asthma_selected_df['ADHECR42'].isnull().sum())

before mean imputation null count :  1166
after mean imputation null count :  0


In [20]:
# Detect outliers
def detect_outliers(asthma_data):
    # Calculate Q1 (25th percentile), Q3 (75th percentile), and IQR for each column
    Q1 = asthma_data.quantile(0.25)
    Q3 = asthma_data.quantile(0.75)
    IQR = Q3 - Q1

    # Calculate lower and upper bounds for outlier detection
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    # Return boolean DataFrame where True indicates an outlier
    return ((asthma_data < lower_bound) | (asthma_data > upper_bound))

# Select numerical columns only
num_cols = asthma_selected_df.drop(['DUPERSID','SEX', 'ASPRIN53', 'RACETHX'],axis=1)

# Detect outliers in the numeric columns
outliers = detect_outliers(num_cols)

# Count the number of outliers for each row
outlier_count = outliers.sum(axis=1)

# Remove rows where more than 3 attributes have outlier values
asthma_selected_cleaned_df = asthma_selected_df[outlier_count <= 3]

# Number of rows left after removing outliers
remaining_rows = asthma_selected_cleaned_df.shape[0]
print(f"Number of tuples left after removing outliers: {remaining_rows}")

Number of tuples left after removing outliers: 2212


In [21]:
# BMINDX col creation
asthma_selected_cleaned_df['BMINDX'] = np.where(asthma_selected_cleaned_df['AGE16X'] > 17, asthma_selected_cleaned_df['BMINDX53'], asthma_selected_cleaned_df['CHBMIX42'])
print(asthma_selected_cleaned_df[['AGE16X','BMINDX53', 'CHBMIX42', 'BMINDX']])

      AGE16X  BMINDX53  CHBMIX42  BMINDX
0       43.0      20.0       NaN    20.0
1       19.0      20.7       NaN    20.7
2        9.0       NaN       NaN     NaN
3       73.0      30.5       NaN    30.5
4       75.0      34.9       NaN    34.9
...      ...       ...       ...     ...
2249    35.0      36.8       NaN    36.8
2250    67.0      28.4       NaN    28.4
2251    51.0      26.1       NaN    26.1
2252    46.0      25.7       NaN    25.7
2253    12.0       NaN      22.9    22.9

[2212 rows x 4 columns]


In [24]:
# Step 1: Calculate the correlation matrix for numeric attributes
correlation_matrix = asthma_selected_cleaned_df.drop(['DUPERSID','SEX', 'ASPRIN53', 'RACETHX'],axis=1).corr()
print(correlation_matrix)

# Step 2: Identify highly correlated pairs (above 0.8 or below -0.8)
# We're setting correlations of 1.0 (self-correlation) to NaN to exclude them
correlation_matrix_filtered = correlation_matrix.where((correlation_matrix != 1) & ((correlation_matrix > 0.8) | (correlation_matrix < -0.8)))

# Step 3: Display the filtered correlation matrix showing only high correlations
print("Highly correlated pairs (correlation > 0.8 or < -0.8):")
print(correlation_matrix_filtered.dropna(how='all'))  # Drop rows where all values are NaN

# Optional: Display attribute pairs with their correlation values as a list
high_corr_pairs = correlation_matrix_filtered.stack()  # Stacks the non-null correlations into pairs
print("\nAttribute pairs with strong correlations:")
print(high_corr_pairs)


          ADAPPT42  ADHECR42    AGE16X  BMINDX53  CHBMIX42  FAMINC16  \
ADAPPT42  1.000000 -0.003790  0.326427  0.095518       NaN -0.076606   
ADHECR42 -0.003790  1.000000  0.024634 -0.018170       NaN  0.033975   
AGE16X    0.326427  0.024634  1.000000  0.047885  0.339359 -0.045189   
BMINDX53  0.095518 -0.018170  0.047885  1.000000  0.920401 -0.135826   
CHBMIX42       NaN       NaN  0.339359  0.920401  1.000000 -0.126124   
FAMINC16 -0.076606  0.033975 -0.045189 -0.135826 -0.126124  1.000000   
WAGEP16X -0.174980  0.043202  0.186034 -0.066508  0.081220  0.384466   
TTLP16X  -0.061124  0.068357  0.414721 -0.065822  0.037219  0.405102   
UNEMP16X -0.023751  0.003396  0.024497 -0.007127       NaN -0.009755   
TOTEXP16  0.352397 -0.002731  0.236018  0.081555  0.085709 -0.081491   
BMINDX    0.095518 -0.016532  0.360121  0.999929  0.998622 -0.140268   

          WAGEP16X   TTLP16X  UNEMP16X  TOTEXP16    BMINDX  
ADAPPT42 -0.174980 -0.061124 -0.023751  0.352397  0.095518  
ADHECR42  0.0

In [23]:
correlation_matrix

ADAPPT42  ADHECR42    AGE16X  BMINDX53  CHBMIX42  FAMINC16  \
ADAPPT42  1.000000 -0.003790  0.326427  0.095518       NaN -0.076606   
ADHECR42 -0.003790  1.000000  0.024634 -0.018170       NaN  0.033975   
AGE16X    0.326427  0.024634  1.000000  0.047885  0.339359 -0.045189   
BMINDX53  0.095518 -0.018170  0.047885  1.000000  0.920401 -0.135826   
CHBMIX42       NaN       NaN  0.339359  0.920401  1.000000 -0.126124   
FAMINC16 -0.076606  0.033975 -0.045189 -0.135826 -0.126124  1.000000   
WAGEP16X -0.174980  0.043202  0.186034 -0.066508  0.081220  0.384466   
TTLP16X  -0.061124  0.068357  0.414721 -0.065822  0.037219  0.405102   
UNEMP16X -0.023751  0.003396  0.024497 -0.007127       NaN -0.009755   
TOTEXP16  0.352397 -0.002731  0.236018  0.081555  0.085709 -0.081491   
BMINDX    0.095518 -0.016532  0.360121  0.999929  0.998622 -0.140268   

          WAGEP16X   TTLP16X  UNEMP16X  TOTEXP16    BMINDX  
ADAPPT42 -0.174980 -0.061124 -0.023751  0.352397  0.095518  
ADHECR42  0.043202  0.068357  0.003396 -0.002731 -0.016532  
AGE16X    0.186034  0.414721  0.024497  0.236018  0.360121  
BMINDX53 -0.066508 -0.065822 -0.007127  0.081555  0.999929  
CHBMIX42  0.081220  0.037219       NaN  0.085709  0.998622  
FAMINC16  0.384466  0.405102 -0.009755 -0.081491 -0.140268  
WAGEP16X  1.000000  0.881652  0.031218 -0.070101  0.082449  
TTLP16X   0.881652  1.000000  0.037048  0.013112  0.134729  
UNEMP16X  0.031218  0.037048  1.000000  0.003182  0.017772  
TOTEXP16 -0.070101  0.013112  0.003182  1.000000  0.130891  
BMINDX    0.082449  0.134729  0.017772  0.130891  1.000000

•	Numerical variables: Since there is high correlation between BMINDX53, BMINDX, and CHBMIX42 variables dropping BMINDX53 and CHBMIX42 and keeping only BMIDX.
•	WAGEP16X and TTLP16X also have high correlation, So dropping WAGEP16X.

In [25]:
# Dropping BMINDX53 and CHBMIX42 since BMINDX now combines both and WAGE16X.
asthma_selected_cleaned_df = asthma_selected_cleaned_df.drop(columns=['CHBMIX42', 'BMINDX53', 'WAGEP16X'])

# Check the remaining columns
asthma_selected_cleaned_df.head()

DUPERSID  SEX  ASPRIN53  ADAPPT42  ADHECR42  AGE16X  FAMINC16  TTLP16X  \
0  10006101    2       2.0       2.0  9.000000    43.0       0.0      0.0   
1  10008102    1       2.0       0.0  8.152574    19.0   45958.0  11432.0   
2  10008106    2       NaN       NaN  8.152574     9.0   45958.0      0.0   
3  10019102    2       1.0       4.0  8.000000    73.0   55921.0  15921.0   
4  10034101    1       2.0       NaN  8.152574    75.0   13332.0  13332.0   

   UNEMP16X  RACETHX  TOTEXP16  BMINDX  
0         0        2        23    20.0  
1         0        2      3418    20.7  
2         0        2      1517     NaN  
3         0        2     15794    30.5  
4         0        3     10340    34.9

In [27]:
from scipy.stats import chi2_contingency

# Define the pairs of columns to test
pairs = [('SEX', 'RACETHX'), ('SEX', 'ASPRIN53'), ('RACETHX', 'ASPRIN53')]

# Loop through each pair and perform the Chi-square test
for var1, var2 in pairs:
    contingency_table = pd.crosstab(asthma_selected_cleaned_df[var1], asthma_selected_cleaned_df[var2])
    chi2, p, dof, expected = chi2_contingency(contingency_table)
    
    print(f"Chi-square Test between {var1} and {var2}")
    print(f"Chi-square statistic: {chi2}")
    print(f"P-value: {p}")
    print(f"Degrees of freedom: {dof}")
    print("Expected frequencies:")
    print(expected)
    print("\n" + "-"*40 + "\n")


Chi-square Test between SEX and RACETHX
Chi-square statistic: 4.992187005113111
P-value: 0.2881000971914933
Degrees of freedom: 4
Expected frequencies:
[[251.57368897 353.17857143 203.61618445  37.79701627  52.83453888]
 [367.42631103 515.82142857 297.38381555  55.20298373  77.16546112]]

----------------------------------------

Chi-square Test between SEX and ASPRIN53
Chi-square statistic: 0.09908461228728255
P-value: 0.7529309126098429
Degrees of freedom: 1
Expected frequencies:
[[119.05006766 353.94993234]
 [252.94993234 752.05006766]]

----------------------------------------

Chi-square Test between RACETHX and ASPRIN53
Chi-square statistic: 11.332635529377734
P-value: 0.023069356132835443
Degrees of freedom: 4
Expected frequencies:
[[ 86.83355886 258.16644114]
 [172.66035183 513.33964817]
 [ 74.50067659 221.49932341]
 [ 16.86332882  50.13667118]
 [ 21.1420839   62.8579161 ]]

----------------------------------------



Categorical Variable: The p-value between RACETHX and ASPRIN53 is 0.023 which indicates significant correlation. Considering, RACETHX gives more information on understanding asthma prevalence and severity retaining that and dropping ASPRIN53.

In [28]:
# Dropping ASPRIN53
asthma_selected_cleaned_df = asthma_selected_cleaned_df.drop(columns=['ASPRIN53'])

# Check the remaining columns
asthma_selected_cleaned_df.head()

DUPERSID  SEX  ADAPPT42  ADHECR42  AGE16X  FAMINC16  TTLP16X  UNEMP16X  \
0  10006101    2       2.0  9.000000    43.0       0.0      0.0         0   
1  10008102    1       0.0  8.152574    19.0   45958.0  11432.0         0   
2  10008106    2       NaN  8.152574     9.0   45958.0      0.0         0   
3  10019102    2       4.0  8.000000    73.0   55921.0  15921.0         0   
4  10034101    1       NaN  8.152574    75.0   13332.0  13332.0         0   

   RACETHX  TOTEXP16  BMINDX  
0        2        23    20.0  
1        2      3418    20.7  
2        2      1517     NaN  
3        2     15794    30.5  
4        3     10340    34.9

In [29]:
# Exclude 'DUPERSID' and other non-numeric categorical attributes
columns_to_exclude = ['DUPERSID', 'SEX', 'RACETHX']  # Add any other categorical variables to this list

# Select numerical columns that are not in the exclusion list
numerical_columns = asthma_selected_cleaned_df.select_dtypes(include=['float64', 'int64']).columns.difference(columns_to_exclude)

# Apply Min-Max scaling to the selected numerical columns
asthma_selected_cleaned_df[numerical_columns] = (asthma_selected_cleaned_df[numerical_columns] - asthma_selected_cleaned_df[numerical_columns].min()) / (asthma_selected_cleaned_df[numerical_columns].max() - asthma_selected_cleaned_df[numerical_columns].min())

# Check the normalized data
print(asthma_selected_cleaned_df[numerical_columns].head())


   ADAPPT42  ADHECR42    AGE16X    BMINDX  FAMINC16  TOTEXP16   TTLP16X  \
0  0.333333  0.900000  0.505882  0.120907  0.000000  0.000055  0.000000   
1  0.000000  0.815257  0.223529  0.129723  0.092757  0.008179  0.051434   
2       NaN  0.815257  0.105882       NaN  0.092757  0.003630  0.000000   
3  0.666667  0.800000  0.858824  0.253149  0.112866  0.037792  0.071631   
4       NaN  0.815257  0.882353  0.308564  0.026908  0.024741  0.059983   

   UNEMP16X  
0       0.0  
1       0.0  
2       0.0  
3       0.0  
4       0.0  


In [30]:
asthma_selected_cleaned_df

DUPERSID  SEX  ADAPPT42  ADHECR42    AGE16X  FAMINC16   TTLP16X  \
0     10006101    2  0.333333  0.900000  0.505882  0.000000  0.000000   
1     10008102    1  0.000000  0.815257  0.223529  0.092757  0.051434   
2     10008106    2       NaN  0.815257  0.105882  0.092757  0.000000   
3     10019102    2  0.666667  0.800000  0.858824  0.112866  0.071631   
4     10034101    1       NaN  0.815257  0.882353  0.026908  0.059983   
...        ...  ...       ...       ...       ...       ...       ...   
2249  80546101    2  0.500000  0.800000  0.411765  0.076696  0.170968   
2250  80550101    2  1.000000  0.900000  0.788235  0.159305  0.001125   
2251  80564101    2  0.166667  1.000000  0.600000  0.195376  0.125976   
2252  80568101    2  0.666667  0.900000  0.541176  0.056771  0.054566   
2253  80568103    1       NaN  0.815257  0.141176  0.056771  0.000000   

      UNEMP16X  RACETHX  TOTEXP16    BMINDX  
0          0.0        2  0.000055  0.120907  
1          0.0        2  0.008179  0.129723  
2          0.0        2  0.003630       NaN  
3          0.0        2  0.037792  0.253149  
4          0.0        3  0.024741  0.308564  
...        ...      ...       ...       ...  
2249       0.0        2  0.006362  0.332494  
2250       0.0        2  0.043606  0.226700  
2251       0.0        1  0.002297  0.197733  
2252       0.0        1  0.020152  0.192695  
2253       0.0        1  0.016412  0.157431  

[2212 rows x 11 columns]

In [31]:
asthma_selected_cleaned_df.to_csv('asthma_data_processed.csv',index=False)